terraform/interpolate.go

```diff
@@ -519,6 +519,16 @@ func (i *Interpolater) interpolateListAttribute(
 	log.Printf("[DEBUG] Interpolating computed list attribute %s (%s)",
 		resourceID, attr)
 
+	// In Terraform's internal dotted representation of list-like attributes, the
+	// ".#" count field is marked as unknown to indicate "this whole list is
+	// unknown". We must honor that meaning here so computed references can be
+	// treated properly during the plan phase.
+	if attr == config.UnknownVariableValue {
+		return attr, nil
+	}
+
+	// Otherwise we gather the values from the list-like attribute and return
+	// them.
```

Comment:
Before the above guard, we'd plow right into the list building below even when the whole list was computed and end up returning an empty list, which was wrong.